src/test/java/com/google/devtools/build/lib/skyframe/ToolchainResolutionFunctionTest.java

```diff
@@ -431,4 +431,45 @@ public void resolve_noMatchingPlatform() throws Exception {
         .hasExceptionThat()
         .isInstanceOf(NoMatchingPlatformException.class);
   }
+
+  @Test
+  public void resolve_forceExecutionPlatform() throws Exception {
+    // This should select platform linux, toolchain extra_toolchain_linux, due to the forced
+    // platform,
+    // even though platform mac is registered first.
```

Comment:
What does the platform registration order mean if the build is explicitly setting the platform to `--platforms=//platforms:linux` anyway?

Reply:
Registration is for execution platform, and --platforms is for target. Updated the comment to hopefully be more clear.